src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -481,10 +512,19 @@ EXTERN_C intptr_t RhGetCurrentThunkContext()
 //  thread        - thread to attach
 extern "C" void PalAttachThread(void* thread)
 {
-#if HAVE_THREAD_LOCAL
-    tls_destructionMonitor.SetThread(thread);
+#ifdef TARGET_LINUX
+    Link* link;
+    link = (Link*)malloc(sizeof(Link));
```

Comment:
`PalAttachThread` is only called once per thread. The linked list is unnecessary

Reply:
Without this linked list, linux builds were failing in previous attempts. I spent some time to figure this one out by experimenting in linux-x64 and linux-musl-x64. I'm not sure how to make it work without a linked list.

Reply:
I have left suggestion with I think it should look like.